File: 771_jewels_in_stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_set<char> _jewels;
        
       for(char c:jewels){
           _jewels.insert(c);
       }
        int count=0;
        
        for(char stone: stones){
            if(_jewels.find(stone) != _jewels.end()){
                ++count;
            }
        }
        return count;
    }
};

File: 771_jewels_in_stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
          Set<Character> _jewels =new HashSet<>();
        
       for(char c:jewels.toCharArray()){
           _jewels.add(c);
       }
        int count=0;
        
        for(char stone: stones.toCharArray()){
            if(_jewels.contains(stone)){
                ++count;
            }
        }
        return count;
    }
}
